### ISSUE TITLE
Is this an typo of the papers's proof?

### ISSUE THREAD

**Issue by umiswing** [HazyResearch/flash-attention#273]
Hello! The paper https://arxiv.org/pdf/2205.14135.pdf prove the algorithm’s correctness in appendix C "Proof of Theorem 1". However, it seems to be an error that $P_{:,:𝑗}V_{:j} = diag(l^{(j)})exp(S_{:,:j} - m^{(j)})V_{:j}$. I think it should be $P_{:,:𝑗}V_{:j} = diag(l^{(j)})^{-1}exp(S_{:,:j} - m^{(j)})V_{:j}$ . And according to the following proof the paper gives, it  should be $diag(l^{(j)})^{-1}$ instead of $diag(l^{(j)})$. Is it a typo?

**Comment by tridao** [comment]
Yes, that's right. There's a typo where it should be $diag(l^{(j)})^{-1}$ instead of $diag(l^{(j)})$. I'll fix it in the next version.

Thanks for the catch!
